### ISSUE TITLE
Ambient light adaptation through webcam snapshots (like Clight) and day/night extra controls.

### ISSUE THREAD

**Issue by tuqueque** [Fushko/gammy#95]
I tried to make a concise and fairly explanatory title, but I'll expand a bit here...

BTW, first of all, what a great application this is! I wish there was something like this for Android (maybe there is, but I'm not aware of). Thanks for the fantastic job you're doing!

Now, here are my "feature requests":

1.- I would love to have independent brightness parameters to control the day/night cycles besides just color temperature. Something like a couple extra sliders below the "Temperature" parameter:

Temperature: 65000 K                          [x]  Auto      ...
     <#-------------------------------------------------->
Brightness multiplier range for night - day: 0.66 - 1.00
     <---------------------------------#=========#>

Hopefully the idea or multipliers makes sense... And of course, no matter the multiplier, brightness won't go below 20%, like the main Range parameter mandates.

2.- This is a more complicated request, but it would be awesome to have!... I'd love to also be able to influence the brightness level based on the ambient/room light detected through the webcam you might have installed (similar to Clight: https://github.com/FedeDP/Clight )

I know that besides being much more complicated to implement (although, Clight is open source and GPL, as you can see)... There's also the issue of how to make it work with all the other parameters. My proposal is something " sub-multiplier range" attached to the day/night cycle multiplier I just proposed:
 
Brightness multiplier range for night - day: 0.66 - 1.00
     <---------------------------------#=========#>
     |__  Ambient light sub-multiplier: 0.50 - 1.00
     <--------------------------#=============#>

And again, no matter the multipliers, brightness won't go below 20%, like the main Range parameter mandates.

Hopefully these messy ASCII doodles and my explanations make sense. Once again, I know the 2nd feature is not trivial to implement and it can be a bit confusing for some users, but I'm just sharing my thoughts on a cool feature like this.

Thanks!

**Comment by Fushko** [comment]
> Hopefully the idea or multipliers makes sense

I can see how it does. A side effect of low temperature is reduced brightness, because you are subtracting blue/green from the screen. It's something I'll consider.

> I'd love to also be able to influence the brightness level based on the ambient/room light detected through the webcam you might have installed similar to Clight

Okay, so you basically want to capture both the screen and the ambient light and adapt the brightness based on both parameters at night. 
This is doable, no doubt. Clight uses Clightd, a daemon I could also use to capture ambient light. However, I don't think I wanna go in that direction for the time being. Having both methods at once sounds a bit overkill, and besides, before I even think of anything else I have to do a substantial rewrite to support multi-monitors. So out of the question for now. You'd be better off asking Clight's developer to implement my screenshot based approach.

**Comment by tuqueque** [comment]
> I can see how it does. A side effect of low temperature is reduced brightness, because you are subtracting blue/green from the screen. It's something I'll consider.

Yes, thank you!... Also, take in mind that my preference is to make the screen darker at night (even with the darkening of the lower temperature), and other users might want to do the opposite. So those multipliers should allow for that... I think two separate multiplier sliders with a single control each one are a better (and more user friendly) way of solving that issue than one slider with two controls:

Brightness multiplier range for day: 1.00
<#------------------------------------------------#>
Brightness multiplier range for night: 0.66
<---------------------------------#----------------->

> Okay, so you basically want to capture both the screen and the ambient light and adapt the brightness based on both parameters at night.

Yes, exactly... Sometimes I work with the room light off even during the day and sometimes I open or close the window, so the lighting in the room changes considerably from day to day or even in the same day.

> This is doable, no doubt. Clight is a daemon I could also use to capture ambient light. However, I don't think I wanna go in that direction for the time being. Having both methods at once sounds a bit overkill, and besides, before I even think of anything else I have to do a substantial rewrite to support multi-monitors. So out of the question for now.

Perfectly understandable and thanks for the feedback!

One last thing I'd like to see in future versions of Gammy... is a control for the Gamma/Contrast depending on the brightness. For low brightness I like to have midtones in the screen relatively brighter (less contrast). In Redshift there's a set of parameters to control the Gamma and it works wonderfully!

Anyway, thank you for your feedback and for your hard work!
